### DISCUSSION TITLE
nm-vars source

### DISCUSSION THREAD

**Discussion by Mutaz94** [metrumresearchgroup/mrgsolve#1082] [General]
Hi @kylebaron 

Is it possible to get source code of `nm-vars`. I saw this behavior:

When I define DADT or A as follow:
```
#define ydot(n) DADT(n)
#define y(n) A(n)
```

It works fine but when I do the same for dose proporities, e.g., `Fn` or `Dn` Things didn't work out:

```
#define bio(n) Fn
#define dur(n) Dn
```
or even
```
#define bio_n Fn
#define dur_n Dn
```

Not sure how was it defined in the header file. Any suggestions?

**Comment by kylebaron** [comment]
Hi @Mutaz94  - 

You can look here:
https://github.com/metrumresearchgroup/mrgsolve/blob/main/inst/base/modelheader.h#L106-L113

But I'd caution against doing this since this isn't part of the public interface at the moment and could go away.

**Comment by Mutaz94** [comment]
Thank you, @kylebaron. This will get the job done. Are you thinking of re-constructing `nm-vars`?

**Reply by kylebaron** [reply]
Hi @Mutaz94  - I'm not planning on it at the moment; but I can't say that it won't change. Anyway ...  the code I shared isn't really `nm-vars`; they all work through those pre-processor definitions.

**Reply by Mutaz94** [reply]
Ahh, that make sense, thank you!!
